docker-compose.yml

```diff
@@ -4,33 +4,98 @@ services:
         build:
             context: .
             dockerfile: ./dockerfiles/Dockerfile
+            target: web-server
         platform: "linux/amd64"
         depends_on:
             - db
             - s3
         ports:
-            - "3000:3000"
-        # for metrics
-        expose: ["3000"]
-        volumes:
-            - "/var/run/docker.sock:/var/run/docker.sock"
-            - ".rustwide-docker:/opt/docsrs/rustwide"
-            - "cratesio-index:/opt/docsrs/prefix/crates.io-index"
+            - "3000:80"
         environment:
-            DOCSRS_RUSTWIDE_WORKSPACE: /opt/docsrs/rustwide
+            DOCSRS_PREFIX: /opt/docsrs/prefix
             DOCSRS_DATABASE_URL: postgresql://cratesfyi:password@db
             DOCSRS_STORAGE_BACKEND: s3
             S3_ENDPOINT: http://s3:9000
             AWS_ACCESS_KEY_ID: cratesfyi
             AWS_SECRET_ACCESS_KEY: secret_key
         env_file:
             - .env
+        healthcheck:
+            test: ["CMD", "curl", "--silent", "--fail", "localhost:80"]
+            interval: 10s
+            timeout: 5s
+            retries: 10
+
+    registry-watcher:
+        build:
+            context: .
+            dockerfile: ./dockerfiles/Dockerfile
+            target: registry-watcher
+        platform: "linux/amd64"
+        depends_on:
+            - db
+        volumes:
+            - "cratesio-index:/opt/docsrs/prefix/crates.io-index"
```

Comment:
I think this is fine regarding the index, important is only that we test that the registry watcher would clone the index when it doesn't find it.

Reply:
Also, assuming that we don't have persistent volumes, currently the index would be re-cloned every time we deploy the container, or restart it. 

So either we should have persistent storage, or perhaps clone the index when building the docker image?

Reply:
Yeah, we probably do want to keep it persistent here to avoid recloning all the time, just have it separated from the builders.